### PR TITLE
Use API v2 for country state and zip in checkout address form.

### PR DESCRIPTION
**Pull request by MatthewKennedy**
Change form field containers from paragraph tags to div tags. #9451

Add id into API V2 for country and states.

Use Ajax call to API V2 in address.js to get state and zip required plus list of states for each country.

### CODE REVIEW

File: api/app/serializers/spree/v2/storefront/state_serializer.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ module Storefront
       class StateSerializer < BaseSerializer
         set_type :state
 
-        attributes :abbr, :name
+        attributes :id, :abbr, :name
```
Comment (damianlegawiec) [inline api/app/serializers/spree/v2/storefront/state_serializer.rb]: @MatthewKennedy you don't need to add `id` as an attribute, it's added automatically by JSON API in a higher node level

Screen Shot 2019-08-05 at 12 22 33 PM

**Review by damianlegawiec — COMMENTED** [review]
@MatthewKennedy 

Thank you for this!

Can we...
1. Fix the CodeClimate issues?
2. Don't change the serializers

In the future we're going to move away from IDs for fetching countries/states, it's more natural to use iso symbols and abbreviations

**Comment by MatthewKennedy** [comment]
@damianlegawiec 

I'll sort that.

File: frontend/app/assets/javascripts/spree/frontend/checkout/address.js [hunk 2]
```diff
@@ -15,8 +15,9 @@ Spree.ready(function ($) {
                 var x_states = []
                 for(var i = 0; i < json.length; i++) {
                     var obj = json[i];
-                    x_states.push(obj.attributes)
+                    x_states.push({ "id" : obj.id, "name" : obj.attributes.name})
                 }
+                console.log(x_states);
```
Comment (damianlegawiec) [inline frontend/app/assets/javascripts/spree/frontend/checkout/address.js]: @MatthewKennedy `console.log` :)

Comment (MatthewKennedy) [reply frontend/app/assets/javascripts/spree/frontend/checkout/address.js]: That’s not the worst of it, when you put an address in and click continue it flashes up billing and shipping address requires country.

I’m assuming that the main ruby code is still expecting an integer for the country_id, and possibly other parts of the  code are expecting an integer country_id such as shipping method and coupon code. 

I’ve had a go but can’t seem to fix it, any pointers?

There’s still a load of tidying up to do for the code climate too.

**Comment by MatthewKennedy** [comment]
Hi @damianlegawiec 

Because the database associates country with state based on ID I'm not sure on where to start making changes to the main code so that the address form works if you use :iso rather in place of  :id in the country selector options.

I've put this pull request into a working state as best I can.

**Comment by damianlegawiec** [comment]
hey @MatthewKennedy thank you for your work. Yes, let's keep using IDs on the frontend, there's going to be more changes in the codebase to support ISO symbols everywhere

**Comment by MatthewKennedy** [comment]
## Behaviour
There are some commented out lines in the address.js file, as it stands the behaviour of a country that does not require a zipcode is such that the zip code field is left visible in the DOM but not required to submit the form.


## NOTE!!
This pull request adds into the API V2 storefront the ability to find by country id. Until the rest of the frontend code is setup to associate states to country by ISO this seems that this is required.

Changed containing elements of forms from `<p>` tags to `<div>` tags as previously discussed.

**Review by damianlegawiec — APPROVED** [review]
👍

**Pull request merged**